kubert/src/server.rs

```diff
@@ -320,10 +322,10 @@ impl TlsKeyPath {
     async fn load_private_key(&self) -> std::io::Result<rustls::PrivateKey> {
         // Open keyfile.
         let pem = tokio::fs::read(&self.0).await?;
-        let mut reader = std::io::BufReader::new(pem.as_slice());
 
         // Load and return a single private key.
-        let keys = rustls_pemfile::rsa_private_keys(&mut reader)?;
+        let keys = rustls_pemfile::pkcs8_private_keys(&mut BufReader::new(pem.as_slice()))
```

Comment:
Looking at the implementation, I think these functions will return an empty vec if the file only contains the opposite key type, not an error. 

https://docs.rs/rustls-pemfile/latest/src/rustls_pemfile/lib.rs.html#75-85

If the vec of pkcs8 keys is empty, we want to look for rsa keys.

Reply:
Ah yeah, great catch!